pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -409,6 +435,529 @@ func TestAddFinalizerToGKENetworkParamSet(t *testing.T) {
 		}
 
 		return finalizerExists, nil
-		//TODO: This should be true when adding gnp finalizer is a part of the reconcile loop.
-	}).Should(gomega.BeFalse(), "GKENetworkParamSet should have the finalizer added.")
+	}).Should(gomega.BeTrue(), "GKENetworkParamSet should have the finalizer added.")
+
+	// simulated client doesnt respect the finalizer and immediately deletes the resource
```

Comment:
does the fake client support finalizers? this is a logic inside the apiserver that handles the Delete process, I'll be surprised that the fakeclient implements it

Reply:
why do we test it then? seems misleading to test a mock specific behavior that is not reflecting the reality

Reply:
I thought it'd be better than nothing, but youre right.

Upon rereviewing, I think we can add a deletion timestamp then update the object to simulate the delete. We can completely remove deletion testing because the deletion tests will be expanded in the next pr, which handles cleanup.

Reply:
fixed